### ISSUE TITLE
update the `Order` check `ValidateUpgradeFields` util method to allow for all supported `Order` types

### ISSUE THREAD

**Issue by charleenfei** [cosmos/ibc-go#3635]
## Summary

In a previous version of the upgradability spec, a channel upgrade could only update the `Ordering` of the channel within a subset of the existing ordering (ie: `Unordered` channels could not be upgraded to `Ordered`, whereas `Ordered` could be updated to both `Unordered` and `Ordered`.) However, because the latest iteration of the spec allows for packets to finish their lifecycles, we can now [upgrade channel ordering to any type of ordering,](https://github.com/cosmos/ibc/pull/967/files#diff-69d7c4ac1d4cdeeabcc757d3a34e85945ebc5613a034a52ebd0769cd8a0fc560R90) provided it is supported by the connection.

The util method for validating upgrade fields should reflect this.
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by damiannolan** [comment]
Partially addressed by removing dead code in #3634. 

We should add the following in a follow up PR to ensure the order is supported by the underlying connection. [ref](https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel/keeper/handshake.go#LL37C2-L52C3)

```go
getVersions := connectionEnd.GetVersions()
if len(getVersions) != 1 {
    return "", nil, errorsmod.Wrapf(
	connectiontypes.ErrInvalidVersion,
	"single version must be negotiated on connection before opening channel, got: %v",
	getVersions,
    )
}

if !connectiontypes.VerifySupportedFeature(getVersions[0], order.String()) {
    return "", nil, errorsmod.Wrapf(
	connectiontypes.ErrInvalidVersion,
	"connection version %s does not support channel ordering: %s",
	getVersions[0], order.String(),
    )
}
```

This should probably be reflected in the spec also if not already. cc @AdityaSripal

**Comment by crodriguezvega** [comment]
Closed by https://github.com/cosmos/ibc-go/pull/3774.
